reana-ui/src/hooks.js

```diff
@@ -30,3 +31,12 @@ export function useSubmit(action) {
 
   return handleSubmit;
 }
+
+/**
+ * React Hook to retrieve the current query string params.
+ * @returns URLSearchParams object
+ */
+export function useQuery() {
+  const { search } = useLocation();
+  return React.useMemo(() => new URLSearchParams(search), [search]);
```

Comment:
It's a pity that we have an old react router version, we could use [useSearchParams hook](https://reactrouter.com/docs/en/v6/api#usesearchparams) in v6.

Have you considered using `query-string` (we have it as a dependency) for parsing the query params instead of `URLSearchParams`? I guess both works in the same way..

Reply:
> It's a pity that we have an old react router version, we could use [useSearchParams hook](https://reactrouter.com/docs/en/v6/api#usesearchparams) in v6.

Looks like there are pretty significant changes between v5 and v6. They even recommend waiting for a backwards compatibility package to be released.

> We recommend waiting for the backwards compatibility package to be released before upgrading apps that have more than a few routes.

> Have you considered using query-string (we have it as a dependency) for parsing the query params instead of URLSearchParams? I guess both works in the same way..

Yes. But I decided to stick with the standard library as they provide the functionality we want. This way is easier to get rid of 3rd party dependencies and make the app lighter, as well as having fewer dependency headaches related to upgrades, vulnerabilities, etc. I guess this would need further discussion and a consensual decision.